src/components/common/AuthRoute.tsx

```diff
@@ -0,0 +1,19 @@
+import { Navigate, Outlet } from 'react-router-dom';
+
+interface Props {
+  redirectRoute: string;
+  needAuth: boolean;
+}
+
+export default function AuthRoute({ needAuth, redirectRoute }: Props) {
+  const token = localStorage.getItem('accessToken');
+
+  if (needAuth) {
+    if (token) {
+      return <Outlet />;
+    } return <Navigate to={redirectRoute} replace />;
+  }
+  if (token) {
+    return <Navigate to={redirectRoute} replace />;
+  } return <Outlet />;
```

Comment:
조금 더 파악이 용이하게 나눠서 쓸 수 있어보여요!
```suggestion
  if (needAuth && token) {
    return <Outlet />;
  }
  if (!needAuth && !token) {
    return <Outlet />;
  }
  return <Navigate to={redirectRoute} replace />;
```

Reply:
넵 참고해서 수정완료했습니다